codegen-lite/src/main/java/software/amazon/awssdk/codegen/lite/regions/ServiceMetadataProviderGenerator.java

```diff
@@ -101,6 +101,10 @@ private CodeBlock regions(Partitions partitions) {
     }
 
     private ClassName serviceMetadataClass(String service) {
+        if ("s3".equals(service)) {
```

Comment:
Might have been better to plumb this through as a customization but we can always do that later.